src/app/UserService.ts

```diff
@@ -1,77 +1,76 @@
-import { DynamoDB } from "aws-sdk";
+import { DynamoDocumentClient } from "src/shared/DynamoDocumentClient";
 import { User } from "./User";
 
 export class UserService {
   private _userTableName: string;
-  private _dbClient: DynamoDB.DocumentClient;
+  private _dbClient: DynamoDocumentClient;
 
   constructor({
     userTableName,
     dbClient,
   }: {
     userTableName: string;
-    dbClient: DynamoDB.DocumentClient;
+    dbClient: DynamoDocumentClient;
   }) {
     this._userTableName = userTableName;
     this._dbClient = dbClient;
   }
 
   get: (email: string) => Promise<User | null> = async (email: string) => {
     const key = { email };
-    const result = await this._dbClient
-      .get({
-        TableName: this._userTableName,
-        Key: key,
-      })
-      .promise();
 
-    const item = result.Item || key;
-    return this.mapUser(item);
+    const result = await this._dbClient.get({
+      TableName: this._userTableName,
+      Key: key,
+    });
+
+    const item = result.Item || { email };
+    return this.mapUser(item as Record<string, unknown>);
   };
 
   registerReportsSectionVisit = async (
     email: string,
     date: Date
   ): Promise<void> => {
-    await this._dbClient
-      .update({
-        TableName: this._userTableName,
-        Key: { email },
-        UpdateExpression:
-          "set reportsSectionLastVisit = :reportsSectionLastVisit",
-        ExpressionAttributeValues: {
-          ":reportsSectionLastVisit": date.toISOString(),
-        },
-      })
-      .promise();
+    await this._dbClient.update({
+      TableName: this._userTableName,
+      Item: {
+        email,
+      },
```

Comment:
In my opinion, the previous name `Key` fitted better with the meaning.

Reply:
I was having a typo with the selected commands, I was using `PutCommand` that is for creating new items in dynamo, and that comand needs a parameter `Item` (we might as well have mapped it). The command needed for updating an existing item was `UpdateCommand` that instead of `Item` receives `Key` as a parameter, so actually all methods will be using `Key`